pkg/services/hivelocity/device/device.go

```diff
@@ -305,15 +328,17 @@ func (s *Service) actionVerifyShutdown(ctx context.Context) actionResult {
 
 	// handle reloading state
 	if isReloading {
-		if provisionCondition.Reason == infrav1.DeviceReloadingReason && hasTimedOut(&provisionCondition.LastTransitionTime, 10*time.Minute) {
-			// TODO: set permanent error with an appropriate label, and go back to the state associate to associate with another device via actionGoBack and return
+		if provisionCondition != nil &&
+			provisionCondition.Reason == infrav1.DeviceReloadingReason &&
+			hasTimedOut(&provisionCondition.LastTransitionTime, 10*time.Minute) {
+			return s.setReloadingTooLongTag(ctx, deviceID, provisionCondition)
 		}
 		conditions.MarkFalse(
 			s.scope.HivelocityCluster,
 			infrav1.DeviceProvisioningSucceededCondition,
 			infrav1.DeviceReloadingReason,
 			clusterv1.ConditionSeverityWarning,
-			"device is reloading",
+			fmt.Sprintf("device is reloading since %s", provisionCondition.LastTransitionTime.Format(time.RFC3339)),
```

Comment:
I think this won't work, as it resets the LastTransitionTime. Correct me if I'm wrong and that time does not depend on the message...

Reply:
@janiskemper I guess you can't do a review like usualy because you created the PR. 

Why do you think think this resets the time?

I removed the change, because provisionCondition could be nil.

Reply:
LastTransitionTime indicates the last time of transition.. The question is how a transition is defined... We would have had to check that if that had been relevant. As it is not, there is no question open :)